packages/glob/src/glob.ts

```diff
@@ -0,0 +1,179 @@
+import * as core from '@actions/core'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as patternHelper from './internal-pattern-helper'
+import {IGlobOptions} from './internal-glob-options'
+import {MatchKind} from './internal-match-kind'
+import {Pattern} from './internal-pattern'
+import {SearchState} from './internal-search-state'
+
+export {IGlobOptions}
+
+/**
+ * Returns files and directories matching the specified glob pattern.
+ *
+ * Order of the results is not guaranteed.
+ */
+export async function glob(
+  pattern: string,
+  options?: IGlobOptions
+): Promise<string[]> {
+  const result: string[] = []
+  for await (const itemPath of globGenerator(pattern, options)) {
+    result.push(itemPath)
+  }
+  return result
+}
+
+export async function* globGenerator(
+  pattern: string,
+  options?: IGlobOptions
+): AsyncGenerator<string, void> {
+  // Set defaults options
+  options = patternHelper.getOptions(options)
+
+  // Parse patterns
+  const patterns: Pattern[] = patternHelper.parse([pattern], options)
+
+  // Push the search paths
+  const stack: SearchState[] = []
+  for (const searchPath of patternHelper.getSearchPaths(patterns)) {
```

Comment:
Can we trace out `searchPath` for debugging, since we might compute the search path base on multiple patterns?